test_run_handler.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wptdashboard
+
+import (
+    "encoding/json"
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "time"
+
+    "appengine"
+    "appengine/datastore"
+)
+
+func testRunHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Yep, sounds good. As a high level description of what this handler does, is it correct to say that it:

 1. Checks that the secret matches, and that everything is OK
 2. Parses the body of the request which should contain JSON
 3. Creates a TestRun entity and stores it in datastore

Right? Also if we want to change away from app engine then obviously we'd have to store the testruns in some other way. But anyway in that case, this whole handler would have to be written in a different way.